compiler/p/codegen/PPCDebug.cpp

```diff
@@ -522,12 +498,7 @@ TR_Debug::print(TR::FILE *pOutFile, TR::PPCTrg1Src1ImmInstruction * instr)
    print(pOutFile, instr->getSource1Register(), TR_WordReg);
    TR::InstOpCode::Mnemonic op =  instr->getOpCodeValue();
 
-   if (op == TR::InstOpCode::subfic || op == TR::InstOpCode::addi || op == TR::InstOpCode::addi2 ||
-       op == TR::InstOpCode::addic  || op == TR::InstOpCode::addic_r ||
-       op == TR::InstOpCode::addis  || op == TR::InstOpCode::mulli)
-      trfprintf(pOutFile, ", %d", (signed short)instr->getSourceImmediate());
-   else
-      trfprintf(pOutFile, ", " "%d", (intptr_t)(int32_t)instr->getSourceImmediate());
+   trfprintf(pOutFile, ", " "%d", (intptr_t)(int32_t)instr->getSourceImmediate());
```

Comment:
Although verbose we should try to use `static_cast` for this, and in general C++ style casting, as it is type safe and will warn us about any changes to the return value type of `getSourceImmediate` in the future. Just thinking ahead as one of the goals for the project is to enable warnings as errors. Not sure if this would be flagged as truncation.

Reply:
This doesn't have to be done here as it is pervasive in the repository, just a note for future.